firmware/shared/controls/DI_driverTorqueRequest.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include <cmath>
+#include <iostream>
+
+namespace ctrl {  
+
+// Templating means this function will work on any types that support addition.
+template <typename T, typename B>
+T DI_driverTorqueRequest(T READY_TO_DRIVE, T DI_Error, B AccelPedPos) {
+    T u1 = 0.0;
+    if((!READY_TO_DRIVE) == 1 || DI_Error == 1){ 
+        return u1;
+    }
+    else{ 
+        //AccelPedPos (n-d Lookup Table)
+        if(AccelPedPos > 1){ return 1.0;}

```

Comment:
this `if / elseif / else` block always returns either 0 or 1, but this is not the behaviour of the original function.

The simulink function just returns the accel_pedal_pos (see the simulink diagram)